geonode/tasks/tasks.py

```diff
@@ -18,14 +18,43 @@
 #
 #########################################################################
 
-from geonode.celery_app import app
-from celery.utils.log import get_task_logger
 from django.conf import settings
 from django.core.mail import send_mail
 
+from django.core.cache import cache
+from contextlib import contextmanager
+
+from celery.five import monotonic
+from celery.utils.log import get_task_logger
+
+from geonode.celery_app import app
+
 logger = get_task_logger(__name__)
 
 
+@contextmanager
+def memcache_lock(lock_id, oid):
+    """
+    ref.
+     http://docs.celeryproject.org/en/latest/tutorials/task-cookbook.html#ensuring-a-task-is-only-executed-one-at-a-time
+    """
+    _exp = settings.CELERY_TASK_LOCK_EXPIRES or 0
+    timeout_at = monotonic() + _exp - 3
+    # cache.add fails if the key already exists
+    status = cache.add(lock_id, oid, settings.CELERY_TASK_LOCK_EXPIRES)
```

Comment:
the cache timeout must be removed (set to None) otherwise it might happen that a new collector runs while the previous is still running, and the new one is able to set the lock since the previous one has expired. This leads to the same problem we're fixing.
The logic in the final block must also be aligned to the use case.

Reply:
@giohappy the settings defualt is None

Reply:
@giohappy I removed the option which might be misleading.